lib/Conversion/HWToBTOR2/HWToBTOR2.cpp

```diff
@@ -0,0 +1,817 @@
+    if (auto ifop = dyn_cast<sv::IfOp>(((Operation *)op)->getParentOp())) {
+      Value en = ifop.getOperand();
+
+      // Generate the implication
+      genImplies(ifop, en, expr);
+
+      // Generate the implies inversion
+      genUnaryOp(op, ifop, notStr, 1);
+    } else {
+      // Generate the expression inversion
+      genUnaryOp(op, expr, notStr, 1);
+    }
+
+    // Genrate the bad btor2 intruction
+    genBad(op);
+  }
+  // Assumptions are converted to a btor2 constraint instruction
+  void visitSV(sv::AssumeOp op) {
+    // Extract the expression that we want our constraint to be about
+    Value expr = op.getExpression();
+    genConstraint(expr);
+  }
+
+  void visitSV(Operation *op) { visitInvalidSV(op); }
+
+  // Once SV Ops are visited, we need to check for seq ops
+  void visitInvalidSV(Operation *op) { visit(op); }
+
+  // Seq operation visitor, that dispatches to other seq ops
+  // Also handles all remaining operations that should be explicitly ignored
+  void visit(Operation *op) {
+    // Typeswitch is used here because other seq types will be supported
+    // like all operations relating to memories and CompRegs
+    TypeSwitch<Operation *, void>(op)
+        .Case<seq::FirRegOp, hw::WireOp>([&](auto expr) { visit(expr); })
+        .Default([&](auto expr) { visitUnsupportedOp(op); });
+  }
+
+  // Firrtl registers generate a state instruction
+  // The final update is also used to generate a set of next btor
+  // instructions
+  void visit(seq::FirRegOp reg) {
+    // Start by retrieving the register's name and width
+    StringRef regName = reg.getName();
+    int64_t w = requireSort(reg.getType());
+
+    // Generate state instruction (represents the register declaration)
+    genState(reg, w, regName);
+
+    // Record the operation for future `next` instruction generation
+    // This is required to model transitions between states (i.e. how a
+    // register's value evolves over time)
+    regOps.push_back(reg);
+  }
+
+  // Ignore all other explicitly mentionned operations
+  // ** Purposefully left empty **
+  void ignore(Operation *op) {}
+
+  // Tail method that handles all operations that weren't handled by previous
+  // visitors. Here we simply make the pass fail or ignore the op
+  void visitUnsupportedOp(Operation *op) {
+    // Check for explicitly ignored ops vs unsupported ops (which cause a
+    // failure)
+    TypeSwitch<Operation *, void>(op)
+        // All explicitly ignored operations are defined here
+        .Case<sv::MacroDefOp, sv::MacroDeclOp, sv::VerbatimOp,
+              sv::VerbatimExprOp, sv::VerbatimExprSEOp, sv::IfOp, sv::IfDefOp,
+              sv::IfDefProceduralOp, sv::AlwaysOp, sv::AlwaysCombOp,
+              sv::AlwaysFFOp, seq::FromClockOp, seq::ClockDivider,
+              seq::ClockGateOp, seq::ClockMuxOp, seq::ConstClockOp,
+              seq::ToClockOp, hw::OutputOp, hw::HWModuleOp>(
+            [&](auto expr) { ignore(op); })
```

Comment:
why are you calling ignore?  Why does ignore exist?

Reply:
`ignore` is just an empty method, I used it to improve readability.